lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -39,11 +39,17 @@ LinearEncodingAttr toLinearEncoding(Attribute layout, ArrayRef<int64_t> shape) {
 }
 
 unsigned getTotalElemsPerThread(Attribute layout, ArrayRef<int64_t> shape) {
+  if (auto distLayout = mlir::dyn_cast<DistributedEncodingTrait>(layout)) {
+    return distLayout.getTotalElemsPerThread(shape);
+  }
```

Comment:
You can even assert that this is indeed a DistributedEncoding as otherwise this function does not make any sense. Either that or directly change the input argument to have the right type.

Reply:
something like:
```c++
unsigned getTotalElemsPerThread(RankedTensorType t) {
  auto layout = cast<DistributedEncodingTrait>(t.getEncoding());
  return layout.getTotalElemsPerThread(t.getShape());
}
```